eZ/Publish/API/Repository/Tests/SearchServiceFulltextEmbedTest.php

```diff
@@ -0,0 +1,181 @@
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $repository = $this->getRepository(false);
+
+        if (false === $repository->getSearchService()->supports(SearchService::CAPABILITY_ADVANCED_FULLTEXT)) {
+            $this->markTestSkipped('Engine says it does not support advance fulltext format');
+        }
+    }
+
+    public function testPrepareContent(): void
+    {
+        $contentService = $this->getRepository()->getContentService();
+        $baseArticleStruct = $this->prepareBaseArticleStruct();
+
+        $slaveArticleStruct = $this->fillSlaveArticleStruct(clone $baseArticleStruct);
+        $slaveArticleContent = $contentService->publishVersion(
+            $this->createContent($slaveArticleStruct)->versionInfo
+        );
+
+        $mainArticleStruct = $this->fillMainArticleStruct(clone $baseArticleStruct, $slaveArticleContent->id);
+        $mainArticleContent = $contentService->publishVersion(
+            $this->createContent($mainArticleStruct)->versionInfo
+        );
+
+        $this->refreshSearch($this->getRepository());
+
+        self::$createdIds = [
+            $slaveArticleContent->id,
+            $mainArticleContent->id,
+        ];
+    }
+
+    /**
+     * @depends testPrepareContent
+     */
+    public function testFulltextContentSearch(): void
+    {
+        $searchService = $this->getRepository()->getSearchService();
+
+        $query = new Query([
+            'query' => new Criterion\FullText(self::SLAVE_ARTICLE_NAME),
+        ]);
+
+        $searchResult = $searchService->findContent($query);
+
+        $this->assertGreaterThanOrEqual(2, $searchResult->totalCount);
+        $this->assertResults($searchResult->searchHits);
+    }
+
+    /**
+     * @depends testPrepareContent
+     */
+    public function testFulltextLocationSearch(): void
+    {
+        $searchService = $this->getRepository()->getSearchService();
+
+        $query = new LocationQuery([
+            'query' => new Criterion\FullText(self::SLAVE_ARTICLE_NAME),
+        ]);
+
+        $searchResult = $searchService->findLocations($query);
+
+        $this->assertGreaterThanOrEqual(2, $searchResult->totalCount);
+        $this->assertResults($searchResult->searchHits);
+    }
+
+    private function prepareBaseArticleStruct(): ContentCreateStruct
+    {
+        $introDocument = new \DOMDocument();
+        $introDocument->loadXML(
+            <<<EOT
+<?xml version="1.0" encoding="UTF-8"?>
+<section xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:ezxhtml="http://ez.no/xmlns/ezpublish/docbook/xhtml" xmlns:ezcustom="http://ez.no/xmlns/ezpublish/docbook/custom" version="5.0-variant ezpublish-1.0">
+<para>some paragraph</para>
+</section>
+EOT
+        );
+
+        $repository = $this->getRepository();
+        $contentType = $repository->getContentTypeService()->loadContentTypeByIdentifier('article');
+
+        /** @var ContentCreateStruct $articleStruct */
```

Comment:
Please use FQCN in variable type hints and remember to rely on API if possible.

Reply:
Changed to FCQN